flytekit/control_plane/launch_plan.py

```diff
@@ -0,0 +1,227 @@
+        in preparation for actual registration to Admin.
+
+        :rtype: flyteidl.admin.launch_plan_pb2.LaunchPlan
+        """
+        # NOTE: does this need to be implemented in the control plane?
+        pass
+
+    @property
+    def id(self) -> _identifier.Identifier:
+        return self._id
+
+    @property
+    def is_scheduled(self) -> bool:
+        if self.entity_metadata.schedule.cron_expression:
+            return True
+        elif self.entity_metadata.schedule.rate and self.entity_metadata.schedule.rate.value:
+            return True
+        elif self.entity_metadata.schedule.cron_schedule and self.entity_metadata.schedule.cron_schedule.schedule:
+            return True
+        else:
+            return False
+
+    @property
+    def auth_role(self) -> _common_models.AuthRole:
+        fixed_auth = super(FlyteLaunchPlan, self).auth_role
+        if fixed_auth is not None and (
+            fixed_auth.assumable_iam_role is not None or fixed_auth.kubernetes_service_account is not None
+        ):
+            return fixed_auth
+
+        assumable_iam_role = _auth_config.ASSUMABLE_IAM_ROLE.get()
+        kubernetes_service_account = _auth_config.KUBERNETES_SERVICE_ACCOUNT.get()
+
+        if not (assumable_iam_role or kubernetes_service_account):
+            _logging.warning(
+                "Using deprecated `role` from config. Please update your config to use `assumable_iam_role` instead"
+            )
+            assumable_iam_role = _sdk_config.ROLE.get()
+        return _common_models.AuthRole(
+            assumable_iam_role=assumable_iam_role,
+            kubernetes_service_account=kubernetes_service_account,
+        )
+
+    @property
+    def workflow_id(self) -> _identifier.Identifier:
+        return self._workflow_id
+
+    @property
+    def interface(self) -> _interface.TypedInterface:
+        """
+        The interface is not technically part of the admin.LaunchPlanSpec in the IDL, however the workflow ID is, and
+        from the workflow ID, fetch will fill in the interface. This is nice because then you can __call__ the=
+        object and get a node.
+        """
+        return self._interface
+
+    @property
+    def resource_type(self) -> _identifier_model.ResourceType:
+        return _identifier_model.ResourceType.LAUNCH_PLAN
+
+    @property
+    def entity_type_text(self) -> str:
+        return "Launch Plan"
+
+    @property
+    def raw_output_data_config(self) -> _common_models.RawOutputDataConfig:
+        raw_output_data_config = super(FlyteLaunchPlan, self).raw_output_data_config
+        if raw_output_data_config is not None and raw_output_data_config.output_location_prefix != "":
+            return raw_output_data_config
+
+        # If it was not set explicitly then let's use the value found in the configuration.
+        return _common_models.RawOutputDataConfig(_auth_config.RAW_OUTPUT_DATA_PREFIX.get())
```

Comment:
nix.  we shouldn't modify what was read from the database based on local settings.